processing/src/main/java/io/druid/query/groupby/GroupByQueryQueryToolChest.java

```diff
@@ -203,24 +205,75 @@ public GroupByQueryQueryToolChest(
           runner,
           context
       );
+      final Sequence<Row> finalizingResults = finalizeSubqueryResults(subqueryResult, subquery);
+      return groupByStrategy.processSubqueryResult(subquery, query, resource, finalizingResults, false);
+    } else {
+      return groupByStrategy.mergeResults(runner, query, context);
+    }
+  }
 
-      final Sequence<Row> finalizingResults;
-      if (QueryContexts.isFinalize(subquery, false)) {
-        finalizingResults = new MappedSequence<>(
-            subqueryResult,
-            makePreComputeManipulatorFn(
-                subquery,
-                MetricManipulatorFns.finalizing()
-            )::apply
-        );
-      } else {
-        finalizingResults = subqueryResult;
-      }
+  private Sequence<Row> getResultsOnPushDown(
+      GroupByStrategy groupByStrategy,
+      GroupByQuery originalQuery,
+      GroupByQueryResource resource,
+      QueryRunner<Row> runner,
+      Map<String, Object> context
+  )
+  {
+    GroupByQuery pushedDownNestedQuery = getNestedQueryToPushDown(originalQuery);
+    Sequence<Row> pushDownQueryResults = groupByStrategy.mergeResults(runner, pushedDownNestedQuery, context);
+    final Sequence<Row> finalizedResults = finalizeSubqueryResults(pushDownQueryResults, pushedDownNestedQuery);
+    return groupByStrategy.processSubqueryResult(pushedDownNestedQuery, originalQuery, resource, finalizedResults, true);
+  }
 
-      return groupByStrategy.processSubqueryResult(subquery, query, resource, finalizingResults);
+  public static GroupByQuery getNestedQueryToPushDown(GroupByQuery query)
```

Comment:
Can be package-private. Also please annotate with `@VisibleForTesting`.

Reply:
This method is similar to [this code](https://github.com/druid-io/druid/blob/master/processing/src/main/java/io/druid/query/groupby/strategy/GroupByStrategyV2.java#L232-L254) in `GroupByStrategyV2.mergeResults()`, and the returned query of this method is immediately passed to that method. Is this needed?

Reply:
Or is this method supposed to return a subquery?